SignalServiceKit/src/Messages/OWSDisappearingMessagesJob.m

```diff
@@ -225,27 +226,25 @@ - (void)becomeConsistentWithConfigurationForMessage:(TSMessage *)message
     [OWSDisappearingMessagesConfiguration fetchOrCreateDefaultWithThreadId:message.uniqueThreadId
                                                                transaction:transaction];
     
-    BOOL changed = NO;
     if (message.expiresInSeconds == 0) {
-        if (disappearingMessagesConfiguration.isEnabled) {
-            changed = YES;
-            DDLogWarn(@"%@ Received remote message which had no expiration set, disabling our expiration to become "
-                      @"consistent.",
-                      self.logTag);
-            disappearingMessagesConfiguration.enabled = NO;
-            [disappearingMessagesConfiguration saveWithTransaction:transaction];
-        }
-    } else if (message.expiresInSeconds != disappearingMessagesConfiguration.durationSeconds) {
```

Comment:
Previously we were only considering duration and not "enabled" state when becoming consistent with disappearing messages.